datahub-web-react/src/app/entity/shared/tabs/Dataset/Queries/QueryCardDetailsMenu.tsx

```diff
@@ -28,9 +29,12 @@ export default function QueryCardDetailsMenu({ urn, onDeleted, index }: Props) {
                     onDeleted?.(urn);
                 }
             })
-            .catch(() => {
-                message.destroy();
-                message.error({ content: 'Failed to delete Query! An unexpected error occurred' });
+            .catch((error) => {
+                handleGraphQLError({
+                    error,
+                    defaultMessage: 'Failed to delete Query! An unexpected error occurred',
+                    permissionMessage: 'Unauthorized to delete Query. Please contact your DataHub administrator.',
+                });
```

Comment:
I vote to make the `permission` message generic with the option to override. `Unauthorized. Please contact your Datahub admin` would suffice.

Reply:
nice I like this

Reply:
+1